lib/pxclient.js

```diff
@@ -57,6 +64,25 @@ class PxClient {
         activity.details = details;
         return activity;
     }
+    async fetchRemoteConfig(config) {
+        try {
+            const callData = {
+                url: `https://sapi-${config.px_app_id}.perimeterx.net/config/`,
+                headers: { 'Authorization': `Bearer ${config.px_remote_config_secret}`, 'Accept-Encoding': '' },
+                timeout: 20000,
+            };
+            const res = await p({ url: callData.url, headers: callData.headers, timeout: callData.timeout, method: 'GET' });
+            const remoteConfigObject = JSON.parse(res.body);
```

Comment:
- Add Validation phase to [check the remote config ID and version](https://github.com/PerimeterX/px-enforcer-spec/wiki/Remote-Configuration#pull_config_protocol:~:text=Verify%20that%20the%20incoming%20config%20version%20and%20config%20ID%20match%20the%20desired%20version%20and%20ID.)
- Add [retry mechanism](https://github.com/PerimeterX/px-enforcer-spec/wiki/Remote-Configuration#:~:text=If%20there%20is%20an%20error%20during%20the%20process%20of%20pulling%20the%20Remote%20Config%2C%20or%20if%20the%20version%20or%20the%20config%20ID%20doesn%27t%20match%2C%20implement%20a%20retry%20mechanism%20with%205%20retries%20at%20intervals%20of%201%20second.%20Report%20each%20error%20to%20the%20HS%20logger.)

Reply:
Done 👍